artemis-protocols/artemis-amqp-protocol/src/main/java/org/apache/activemq/artemis/protocol/amqp/connect/mirror/AMQPMirrorControllerTarget.java

```diff
@@ -177,6 +180,48 @@ public void onError(int errorCode, String errorMessage) {
 
    private AckManager ackManager;
 
+   /** This method will wait both replication and storage to finish their current operations. */
+   public void flush() {
+      CountDownLatch latch = new CountDownLatch(1);
+      connection.runNow(() -> {
+         OperationContext oldContext = OperationContextImpl.getContext();
+         try {
+            OperationContextImpl.setContext(mirrorContext);
+            mirrorContext.executeOnCompletion(new IOCallback() {
+               @Override
+               public void done() {
+                  latch.countDown();
+               }
+
+               @Override
+               public void onError(int errorCode, String errorMessage) {
+                  logger.warn("error code = {} / message = {}", errorCode, errorMessage);
```

Comment:
The log message isnt particularly descriptive, should be clearer whats going on and the impact.